### PR TITLE
fix(traints): ensure an error is returned to the caller if a catalog cannot be found

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

File: pkg/trait/camel.go [hunk 1]
```diff
@@ -51,6 +55,9 @@ func (t *camelTrait) Apply(e *Environment) error {
 			if err != nil {
 				return err
 			}
+			if c == nil {
+				return fmt.Errorf("unable to find catalgo for: %s", version)
```
Comment (astefanutti) [inline pkg/trait/camel.go]: nit: typo

File: pkg/trait/camel.go [hunk 2]
```diff
@@ -70,6 +77,9 @@ func (t *camelTrait) Apply(e *Environment) error {
...
```
Comment (astefanutti) [inline pkg/trait/camel.go]: nit: typo

File: pkg/trait/camel.go (hunk 1 above)
Comment (lburgazzoli) [reply pkg/trait/camel.go]: done

File: pkg/trait/camel.go (hunk 2 above)
Comment (lburgazzoli) [reply pkg/trait/camel.go]: done

**Pull request merged**